.changeset/shiny-bushes-stare.md

```diff
@@ -0,0 +1,5 @@
+---
+'posthog-js': major
```

Comment:
i'm tempted to say this is a fix

the behaviour of the sdk is wrong
and now it would be right

this is only breaking if you are coupled to the incorrect behaviour

when we merge the change, we open a PR against posthog main, that updates the version there and when that is released it updates our CDN. ~five minutes later everyone using `array.js` in the snippet gets this code when they next refresh

i think importantly for me we're not changing the type signature which remains`boolean | undefined`  we just would never previously have ever returned undefined

(will catch up in slack or synchronously about the avoiding breaking breaking changes)

Reply:
Thanks for the feedback and the additional context, it's super helpful!

I agree that this can be a patch considering:
- the return type is unchanged
- the documentation has already described the correct return values

It may still affect the behavior of any clients who relied on the incorrect behavior depending on how they implemented it. It might be a little odd, but I suppose it _is_ possible, e.g.:
```ts
const enabled = posthog.isFeatureEnabled('foo') ?? true
```
or
```ts
if (posthog.isFeatureEnabled('foo') !== false) { ... }
```

I don't yet have a great sense of whether this is an acceptable amount of backwards incompatibility in exchange for the bug fix. Perhaps it's safer to wait until we have better facilities for end-users to update or rollback their snippet.

Reply:
i haven't looked at the implementation but @robbie-c added `defaults` to the config if i remember the name correctly which was intended to let us version things
so you could be super cautious and add a new "default" so that older clients keep the old behaviour and new clients get the new behaviour

it's such a small change that i'm tempted to yolo but a call for the FF team tbh

i added "request changes" just cos i didn't have time to read the PR and saw major change and panicked 🤣